### PR TITLE
Update glossary defs for components, templates, and views

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Definitions use "component" to mean "component class".

Issue Number: N/A

## What is the new behavior?
Clarify that `@Component` directive associates a data class with a template.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/glossary.md [hunk 1]
```diff
@@ -964,10 +964,10 @@ To learn more, see [Angular Universal: server-side rendering](guide/universal).
 
 The smallest grouping of display elements that can be created and destroyed together.
 Angular renders a view under the control of one or more [directives](#directive),
-especially [component](#component) directives and their companion [templates](#template).
+especially [component](#component) directives, which associate data classes with display [templates](#template).
```
Comment (kapunahelewong) [inline aio/content/guide/glossary.md]: I find this wording harder to parse than how it was before.

**Review by kapunahelewong — COMMENTED** [review]
Looks good. Just one small thing.

File: aio/content/guide/glossary.md (hunk 1 above)
Comment (jbogarthyde) [reply aio/content/guide/glossary.md]: This is what you pointed out to me -- that a "component" (the way we generally use the term) is actually the combination of the decorated class (which holds the data) and the associated template.  It's the `@Component` directive that defines the view by making that association.

Can you think of a better way to make that more clear and accurate?

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**